packages/app/controllers/user/data.js

```diff
@@ -0,0 +1,46 @@
+const { STATUS_CODES } = require("http");
+const UserService = require("../../services/User");
+const KeyService = require("../../services/Keys");
+const SubscriptionService = require("../../services/Subscription");
+
+async function getUserDataController(req, res, next) {
+  try {
+    const userService = new UserService();
+    const keyService = new KeyService();
+    const subscriptionService = new SubscriptionService();
+    
+    const { userId } = req.userData;
+    const user = await userService.getUser(userId);
+    if (!user) {
+      return res.status(404).json({
+        statusCode: 404,
+        error: STATUS_CODES[404],
+        message: "User document not found",
+      });
+    }
+
+    const data = {};
+
+    const [subscriptionData, userKeys] = await Promise.allSettled([subscriptionService.getSubscription(user.subscription_id), 
```

Comment:
try removing this logic from here.

Reply:
Simplified logic and wrote early return to prevent incorrect user data being sent.